modules/libsodium/1.0.19/overlay/BUILD.bazel

```diff
@@ -0,0 +1,45 @@
+"""Bazel build file for libsodium."""
+
+load("@rules_cc//cc:defs.bzl", "cc_library")
+load("@rules_license//rules:license.bzl", "license")
+load("//bzl:defs.bzl", "gen_version_hdr")
+
+package(default_applicable_licenses = [":license"])
+
+license(
+    name = "license",
+    license_kinds = ["@rules_license//licenses/spdx:ISC"],
+    license_text = "LICENSE",
+)
+
+REPO_ROOT = package_relative_label(":BUILD.bazel").workspace_root
+
+INCLUDE_PREFIX = REPO_ROOT if REPO_ROOT else "."
+
+gen_version_hdr(
+    name = "version_hdr",
+)
+
+cc_library(
+    name = "libsodium",
+    srcs = glob(["src/**/*.c"]),
+    hdrs = glob(["src/**/*.h"]) + [":version_hdr"],
+    copts = [
+        # Add this so version.c can include version.h without the sodium sub-directory.
+        "-I$(GENDIR)/%s/sodium" % (INCLUDE_PREFIX,),
+        "-I%s/src/libsodium/include/sodium" % (INCLUDE_PREFIX,),
+    ] + select({
+        "@platforms//os:windows": [],
+        "//conditions:default": [
+            "-Wno-unused-but-set-variable",
+            "-Wno-unused-function",
+            "-Wno-unknown-pragmas",
+            "-Wno-unused-variable",
+        ],
+    }),
+    # This is defined by the configure script and is used in sodium.h, so unfortunately we have to
```

Comment:
Could you file an issue with libsodium? This is a problematic pattern even without Bazel.

Reply:
This build file is quite old, so I looked it over. Turns out my comment from years past was wrong. That define is not exposed in public headers, so I was able to just do `copts = ["-DCONFIGURED=1"]` instead